### PR TITLE
Fix golint errors in pkg/controller/service

### PR DESCRIPTION
**Pull request by ghost**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Fix golint errors in pgk/controller/service (part of broader goling effort)

**Which issue(s) this PR fixes**:

Related to https://github.com/kubernetes/kubernetes/issues/68026

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by neolit123** [comment]
https://github.com/kubernetes/kubernetes/search?q=ServiceController&unscoped_q=ServiceController

^ this search reveals a lot of usages of the `ServiceController` object that need to be changed too.

**Comment by cristiandima** [comment]
> https://github.com/kubernetes/kubernetes/search?q=ServiceController&unscoped_q=ServiceController
> 
> ^ this search reveals a lot of usages of the `ServiceController` object that need to be changed too.

None of the other usages seem to be within the scope of this PR and issue. https://github.com/kubernetes/kubernetes/issues/68026 is about fixing golint errors package by package across the entire repo. This PR does that for the one package, `pkg/controller/service`.

The specific golint issue identified here is that where object names begin with the name of the package to which they belong - which leads to calls like `service.ServiceController` instead of the cleaner `service.Controller`. This specific issue does not seem to apply to any of the other usages.

**Comment by cblecker** [comment]
@cristiandima You are changing the name of an exported function. This means that any consumers need to change the function they are calling too. This PR would break those consumers, without also fixing all the places where it needs to be fixed.

This PR cannot be accepted in its current state.

**Comment by cristiandima** [comment]
@cblecker only the struct name has been changed and that is only used inside the package. No exported function or method has been changed. I ran the scripts as instructed in the issue with no errors. My compiler does not seem to complain either. I don't understand what makes this PR unacceptable in its current state.

**Comment by danwinship** [comment]
> only the struct name has been changed and that is only used inside the package

Right, `cmd/kube-controller-manager` and `cmd/cloud-controller-manager` call the `New()` function in this file, but they just store the result in a local variable without ever explicitly mentioning its type.


Maybe rename `service_controller.go` → `controller.go` to match the type though (and likewise the test file)?

**Comment by cristiandima** [comment]
> Maybe rename service_controller.go → controller.go to match the type though (and likewise the test file)?

Yes, done that just now.

/assign @andrewsykim @cblecker @timothysc

**Comment by timothysc** [comment]
/ok-to-test 

This needs a rebase, also it the struct is now used why is it exported, why not make it local/private to the package?

**Comment by cristiandima** [comment]
> /ok-to-test
> 
> This needs a rebase, also it the struct is now used why is it exported, why not make it local/private to the package?

The type itself is not used externally at this time, but instances of the struct are, via the `New()` function (and golint does not like private structs as return types). The pattern across controller packages also seems to be to have these controller structs exported.

The golint error we would introduce if this was made private:

```
pkg/controller/service/controller.go:131:4: exported func New returns unexported type *service.controller, which can be annoying to use
```

So maybe keep it as it is (exported/public)?

**Comment by oomichi** [comment]
verify-bazel which is a part of pull-kubernetes-verify job is failed.
Could you run `./hack/update-bazel.sh` and apply the change to the PR?

**Comment by oomichi** [comment]
Unfortunately rebase is needed now.

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Comment by cblecker** [comment]
/approve
for hack/

File: pkg/controller/service/BUILD [hunk 1]
```diff
@@ -3,9 +3,9 @@ load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
 go_library(
     name = "go_default_library",
     srcs = [
+        "controller.go",
         "doc.go",
         "patch.go",
-        "controller.go",
```
Comment (andrewsykim) [inline pkg/controller/service/BUILD]: minor comment -- can you squash this commit with the previous one since this is technically part of the file rename?

Comment (andrewsykim) [reply pkg/controller/service/BUILD]: incase it's not clear via github UI, squash last commit with 2nd last commit

Comment (cristiandima) [reply pkg/controller/service/BUILD]: commits squashed

**Pull request merged**